Source/ToolCore/Command/ProjectCmd.cpp

```diff
@@ -0,0 +1,133 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#include <Atomic/Core/StringUtils.h>
+#include <Atomic/IO/Log.h>
+#include <Atomic/IO/File.h>
+#include <Atomic/IO/FileSystem.h>
+
+#include "../ToolSystem.h"
+#include "../Project/ProjectFile.h"
+#include "../Project/ProjectEvents.h"
+
+#include "ProjectCmd.h"
+
+#include <Poco/File.h>
+
+#define  PROJECTCMD_CACHE_CLEAN 0x1 // Cache command first cleans current cache
+
+namespace ToolCore
+{
+
+ProjectCmd::ProjectCmd(Context* context) : 
+    Command(context),
+    requiresProjectLoad_(false),
+    options_(0)
+{
+
+}
+
+ProjectCmd::~ProjectCmd()
+{
+
+}
+
+// usage: project <projectPath> [cache (-clean)]
+bool ProjectCmd::Parse(const Vector<String>& arguments, unsigned startIndex, String& errorMsg)
+{
+    String argument = arguments[startIndex].ToLower();
+    if (argument != "project")
+    {
+        errorMsg = "Unable to parse project command";
+        return false;
+    }
+
+    command_ = startIndex + 2 < arguments.Size() ? arguments[startIndex + 2] : String::EMPTY;
+    if (command_ == "cache")
+    {
+        requiresProjectLoad_ = true;
+        SubscribeToEvent(E_PROJECTLOADED, ATOMIC_HANDLER(ProjectCmd, HandleProjectLoaded));
+
+        for (unsigned i = startIndex + 3; i < arguments.Size(); i++)
+        {
+            if (arguments[i].Length() > 1 && arguments[i][0] == '-')
+            {
+                String argument = arguments[i].Substring(1).ToLower();
+                String value = i + 1 < arguments.Size() ? arguments[i + 1] : String::EMPTY;
+
+                if (argument == "clean")
+                {
+                    options_ = options_ | PROJECTCMD_CACHE_CLEAN;
+                    SubscribeToEvent(E_PROJECTBEGINLOAD, ATOMIC_HANDLER(ProjectCmd, HandleProjectBeginLoad));
+                    i++;
+                }
+            }
+        }
+    }
+    else
+    {
+        errorMsg = "Unknown project command";
+        return false;
+    }
+
+    projectPath_ = startIndex + 1 < arguments.Size() ? arguments[startIndex + 1] : String::EMPTY;
+
+    return true;
+}
+
+void ProjectCmd::HandleProjectBeginLoad(StringHash eventType, VariantMap& eventData)
+{
+    if (options_ & PROJECTCMD_CACHE_CLEAN)
+    {
+        String cachePath = GetPath(eventData[ProjectBeginLoad::P_PROJECTPATH].GetString());
+        cachePath = AddTrailingSlash(cachePath) + "Cache";
+
+        FileSystem* fileSystem = GetSubsystem<FileSystem>();
+        if (!fileSystem->RemoveDir(cachePath, true))
+            Error("Cache clean failed");
```

Comment:
For future reference, I think we can adopt brackets surrounding single line conditionals, I like them better too, and in cases of macros they can avoid funky code expansion errors (we'll setup an Atomic coding standard)